src/napari_graph/directed_graph.py

```diff
@@ -0,0 +1,498 @@
+            edges,
+            self._empty_edge_idx,
+            self._n_edges,
+            self._node2edges,
+            self._node2tgt_edges,
+        )
+
+    def edges(
+        self, nodes: Optional[ArrayLike] = None, mode: str = 'indices'
+    ) -> Union[List[np.ndarray], np.ndarray]:
+        """`source_edges` alias"""
+        return self.target_edges(nodes, mode)
+
+    def out_edges(
+        self, nodes: Optional[ArrayLike] = None, mode: str = 'indices'
+    ) -> Union[List[np.ndarray], np.ndarray]:
+        """`source_edges` alias"""
+        return self.source_edges(nodes, mode)
+
+    def source_edges(
+        self, nodes: Optional[ArrayLike] = None, mode: str = 'indices'
+    ) -> Union[List[np.ndarray], np.ndarray]:
+        """Return the source edges (outgoing) of the given nodes.
+
+        If no nodes are provided, all source edges are returned.
+
+        Parameters
+        ----------
+        nodes : Optional[ArrayLike], optional
+            Node indices, by default None
+        mode : str
+            Type of data queried from the edges. For example, `indices` or
+            `coords`.
+
+        Returns
+        -------
+        List[np.ndarray]
+            List of (N_i) x 2 x D arrays, where N_i is the number of edges at
+            the ith node.  D is the dimensionality of `coords` when
+            mode == `coords` and is ignored when mode == `indices`. N_i
+            dimension is ignored when N_i is 1.
+        """
+        return self._iterate_edges_generic(
+            nodes,
+            node2edges=self._node2edges,
+            iterate_edges_func=_iterate_directed_source_edges,
+            mode=mode,
+        )
+
+    def in_edges(
+        self, nodes: Optional[ArrayLike] = None, mode: str = 'indices'
+    ) -> Union[List[np.ndarray], np.ndarray]:
+        """`target_edges` alias"""
+        return self.target_edges(nodes, mode)
+
+    def target_edges(
+        self, nodes: Optional[ArrayLike] = None, mode: str = 'indices'
+    ) -> Union[List[np.ndarray], np.ndarray]:
+        """Return the target edges (incoming) of the given nodes.
+
+        If no nodes are provided, all target edges are returned.
+
+        Parameters
+        ----------
+        nodes : Optional[ArrayLike], optional
+            Node indices, by default None
+        mode : str
+            Type of data queried from the edges. For example, `indices` or
+            `coords`.
+
+        Returns
+        -------
+        List[np.ndarray]
+            List of (N_i) x 2 x D arrays, where N_i is the number of edges at
+            the ith node.  D is the dimensionality of `coords` when
+            mode == `coords` and it's ignored when mode == `indices`. N_i
+            dimension is ignored when N_i is 1.
+        """
+        return self._iterate_edges_generic(
+            nodes,
+            node2edges=self._node2tgt_edges,
+            iterate_edges_func=_iterate_directed_target_edges,
+            mode=mode,
+        )
```

Comment:
For maintenance reasons, personally, I'd prefer to not introduce aliases. Don't have strong opinions regarding source/target vs in/out, though.